service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -552,4 +558,16 @@ private ApiCloneControlledGcpBigQueryDatasetResult fetchCloneBigQueryDatasetResu
   private AuthenticatedUserRequest getAuthenticatedInfo() {
     return authenticatedUserRequestFactory.from(request);
   }
+
+  private AuthenticatedUserRequest getPetRequest(UUID workspaceId) {
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    return petSaService
+        .getWorkspacePetCredentials(workspaceId, userRequest)
+        .orElseThrow(
+            () ->
+                new BadRequestException(
```

Comment:
nit: Not sure, but would `InternalServerError` be more appropriate here? If we're trying to clone a bucket, then presumably we already have a backing GCP project and pet SA for this workspace. I usually think of 400 as an error the caller should fix, and if they've already created a GCP project then I'm not sure what action they could take to fix this.